### PR TITLE
New job: Orderly

### PR DESCRIPTION
**Pull request by warior4356**
## What Does This PR Do
This PR adds a new job called orderly.
The orderly's job description is as follows. Perform minor first aid, clean the medbay, restrain resisting patients, and remove trespassers. To do this they spawn with soap, spray cleaner, and a version of CQC that has no harm based moves and can only be used in the medbay. To prevent them from just being doctor+, they do not get surgery access.

## Why It's Good For The Game
Compared to basically any other department, people don't respect the access requirements of or understand that it is trespassing to go in medical without being invited by someone who works there. Similarly stocking up on medicine is meta and no one treats this like the theft it is. On top of the trespassing and theft, ask any veteran medbay player and they will tell you stories about dying patients who are resisting instead of getting treated. The common response to this is security should be dealing with this, but in most cases this both something they don't care about and are too busy to deal with it.

The other problem medbay often has is the frankly massive amount of blood everywhere, however as infections are a thing and medbay should be kept clean it either means harassing the janitor or getting soap or cleaner with a doctor or the CMO cleaning the floor instead of treating people.

In both cases, the orderly exists to solve this problem.

To preemptively address some common worries:

- This role will cause valid hunting. - As their only ability to deal with antags only works in the medbay, this shouldn't be an issue.
- Medbay is already crowded enough. - In my eyes most of that crowding comes from people treating medbay like it is all access. This new job should help with that.
- This job will just be doctor+ - The orderly does not get surgery access which should help restrict them to only doing minor treatment. 

Finally, while the exact number would be up to the head admins, I personally would give this a karma cost of 5-15. If head admins and maintainers are in favor of this, I will edit the PR to add it.

## Sample SOP
1.    Though not mandatory, it is recommended that Orderlies wear Sterile Masks and Latex/Nitrile gloves when handling patients. This Guideline becomes mandatory during Viral Outbreaks;
2.    Orderlies must maintain the entirety of Medbay in an hygienic state. This includes, but is not limited to, cleaning organic residue, fluids and corpses;
3.    Orderlies are not permitted to leave Medbay to perform recreational activities if there are unattended patients requiring treatment;
4.    Orderlies are only to perform non superficial treatment when there are no other medical personnel available to do it. This includes surgery, cloning, the sleepers and cryo;
5.    Orderlies are to man the front desk whenever possible and make sure all non medical personal in medbay are being escorted by a member of medical at all times. If someone is unescorted they should be asked to wait in the lobby and if they refuse, ejected;
6.    Orderlies are allowed to retrieve stolen items while the perpetrator is still in medbay and may restrain them while waiting for security to arrive. They are not allowed to take perpetrators to security, nor are they allowed to “arrest” anyone in or out of medbay;
7.    Orderlies are to restrain disruptive and resisting patients so they can be treated. They should never chase or restrain someone outside of medbay. If they do, they are guilty of assault like any other crew member;
8.    As a reminder, the orderly is not sec and should not be arresting anyone. They should only be using their CQC on someone if they are actively trespassing, resisting, or stealing in medbay;

## Images of changes
Outfit, Backpack, and Loadout
image
Locker and spawn points on Box Station
image
Locker and spawn points on Meta Station
image
Locker and spawn points on Delta Station
image

## Changelog
:cl:
add: Added Orderly Job to Medical
/:cl:

### CODE REVIEW

File: code/game/jobs/job/medical.dm [hunk 1]
```diff
@@ -315,3 +315,49 @@
...
+	title = "Orderly"
+	flag = JOB_ORDERLY
+	department_flag = JOBCAT_MEDSCI
+	total_positions = 1
+	spawn_positions = 1
+	is_medical = 1
+	supervisors = "the chief medical officer"
+	department_head = list("Chief Medical Officer")
+	selection_color = "#ffeef0"
+	access = list(ACCESS_MEDICAL, ACCESS_MORGUE, ACCESS_SURGERY, ACCESS_CHEMISTRY, ACCESS_VIROLOGY, ACCESS_GENETICS, ACCESS_MINERAL_STOREROOM)
```
Comment (SteelSlayer) [inline code/game/jobs/job/medical.dm:329]: You said they don't have surgery access but they have it here in their access list. Was that a mistake?

Comment (warior4356) [reply code/game/jobs/job/medical.dm:329]: No. There are two access lists.
Access is for low pop.
Minimal access is for high pop.

Comment (SteelSlayer) [reply code/game/jobs/job/medical.dm:329]: I see. All good then.

**Comment by AffectedArc07** [comment]
Few things.

I said on the discord this feels like a solution looking for a problem. The only rounds where medbay is unbearable are nuke ops and blobs, both of which tend to end within an hour. Most rounds I have played in medical have been fine, and there's usually a member of security in medbay being treated. What's stopping you from just halting their treatment for a second to say "hey mate can you make that guy fuck off please?"

Removing surgery access from them won't do anything, they will just go straight to the hop and ask for it because of "incase of high medical demand!", and as for the "people thinking medbay is all access" part, how about CMOs stop setting the doors to emergency access instead of getting what is essentially a medical bouncer.

**Comment by KozackWolf** [comment]
Oh god, I main medical, and this would help so much. 

As for the CMO's setting the doors to emergency... I don't see that unless it's a war dec, blob, or wiznerd.(early end round antags)

Trying to take care of a patient, people run in, push past you and cause your action to reset

People trying to treat the person themselves, while you are busy helping another person who is in crit, when they have a "booboo" but want meds for it

People stealing meds on the tables to treat themselves.

People climbing over the tables to get into chem.

Botany stealing chem machines.

Genetics being overrun for "Cute genes" like "smol" making cloning over-run.

People just lounging in medical thinking it's a "hangout" then an emergency happens and they are in the way

Idiots bringing IPC's in and insisting you fix them.

Implant beggars who have no reason to have impants

Greytide in medical.

Need someone who can clean so I don't have to, I can save people.

Medical is a super highway of traffic for people who don't belong in there.

So many issues with trespassing in medical... it's a nightmare. A new job just to handle that sounds amazing...

**Comment by AffectedArc07** [comment]
Also, forgot to mention this before.

Due to this job having a DB entry flag, this PR **cannot** be testmerged, so if that was in your intentions, you'll need to find a workaround.

**Comment by ExusA** [comment]
This is quite cool, but as someone who is both a doctor and a patient quite a bit, I'm not sure this'll improve the experience for most people. I feel there's going to be a lot of cases of the orderly getting bored and ejecting people that don't need to be, disrupting treatment and generally making a nuisance of themselves and making medbay less cooperative with the rest of the crew.
A similar idea that might solve the problems medbay has is a receptionist, who would work at the medbay desk and has a HoP-style ticket machine and waiting room, possibly even give appointments and that sort of thing for RP value. This'd keep them out of the way, and also keep the influx of med-stealing maniacs low and has some possible RP benefit, where I think a bouncer might not. Just thoughts though, I agree something like this has been long in the making.

**Comment by TDSSS** [comment]
In regards to 
> they will just go straight to the hop and ask for it because of "incase of high medical demand!"
There are med at jobs without surgery access and I haven't seen those go ask for surgery access from.the HoP "just in case"

**Comment by TheMaveran** [comment]
I don’t see why the tried and true Ether syringe gun can’t do this guys job.

**Comment by warior4356** [comment]
> I don’t see why the tried and true Ether syringe gun can’t do this guys job.

Because ether takes 82 seconds to knock someone out.

Neuro-toxin (the drink) is better, at only 26 seconds, but is still far too slow.

**Comment by KozackWolf** [comment]
Syringe guns get swiped too much and you never know who has them on them for it to be reliable.

If it becomes an issue of worrying about abuse, karma lock it at 5 or 10 karma.

**Comment by TheMaveran** [comment]
Is he here to deal with antags or tiders? If it’s tiders then I fail to see why ether isn’t gonna work. If it’s antagonists, just call security. This is a solution in search of a problem that doesn’t exist, or at the least, I have failed to notice while in medical.

**Comment by KozackWolf** [comment]
> 
> 
> Is he here to deal with antags or tiders? If it’s tiders then I fail to see why ether isn’t gonna work. If it’s antagonists, just call security. This is a solution in search of a problem that doesn’t exist, or at the least, I have failed to notice while in medical.

Most people, unless they are medical mains, and play during high pop, would fail to notice it.

**Comment by procdrone** [comment]
So the issues mentioned:

>Medbay is often bloody and not cleaned
>Trespassing is rampant in Medbay.

The first point, adding a custom job to clean removes the janitor relevance even more, and given current politic of "interdepartmental cooperation" this just doesn't fly.

Besides, do give the poor janitor do his thing. Janiborgs are their horror enough.

As for trespasser issues - there are two solutions. Make medbay entrance airlocked, like science entrance. This will prevent most people running past open doors. Remove one direction exit. Have medical actually man the medbay lobby and let people in and out.

Medbay staff should also make effort to report people who trespass in medical without injuries to security.

Second, more grounded solution (grounded i mean most other codebases include it) - return the medical security officer back into medbay.

I dont feel like adding a completely new job here will be helpful to fix these issues, and will only generate more problems when player will eventually become bored.

**Comment by Varghh** [comment]
I have mained both medical and security since i started playing and please, this is such a brilliant idea to help the people who play medbay and overall enjoyment of the department a lot.
I have seen WAY too many times civilians just walking in, stealing defribs, medkits or any drug around and just walking out, there are too many situation where even if you call for security they are too busy dealing with the current antag so they can't just answer for a shitter going around medbay, plus this makes the CMO have the responsibility of going around and batoning people out of the medbay instead of taking care of his real job, making sure the place has everything it needs and doctors are doing his job.
I'm all up for this job, the doctor+ idea can be used for paramedic and virologist if they get surgery access and those things don't happen most of the time, so i don't see why that's a good argument.
The validhunting thing would not work because the CQC don't work outside medbay, but if you decide to do something inside it, you need to remember the bouncer may be around, same thing for some officers who literally ask for medbay access and all they do is protect the place.
This is a great idea and honestly i don't see why it should not get merged, or at least tested a few times.
This is a +1 for me and i truly hope people could see all the positive aspect instead of the minors "what if" they are trowing around.

**Comment by AffectedArc07** [comment]
This was discussed at length in mentor chat over discord (Spanning from 21:04 to 00:33 GMT), and here is a brief summary of points

- This requires snowflaking of martial arts code
- This makes any medical doctor target much harder to kill since this job will descend into validhunting, with a weapon that cant be confiscated
- You saying `"It's only possible because space law is already disregarded for medbay"` is not only incorrect, it also comes off as a really pushy way in arguing for this role
- Any argument about validhunting was very quickly shut down with `"They were tresspassing"`
- Quite a few admins spoke out about this, and to summarise what one said, `"Medbay vendors never really get stolen, and of the 4, one is usually handed to the brig physician."`
- When someone questioned one thing, you immediately linked them a comment, then said `"Well how would you fix it then [name]? If you dont like my solution, lets take a look at yours"`, which doesnt come off as that nice of a tone
- People self healing then walking out of medbay was replied to with `"People shouldnt self heal. Full stop."`
- Botany stealing chem machines was a bad point to bring up, largely because A, that never happens, and B, thats more of securities job to deal with
- This entire job is just bootleg security 
- The name orderly really doesnt fit
- You seem extremely hyperbolic of what is seemingly a non issue for a lot of players

We have security to deal with these things. A lot of the time medbay doesnt have tresspassers who mean harm, they just walk in, heal themselves (Or get a friend to), then walk out. Theft from science and engineering occurs a lot more than this, so why does medical require its own bootleg security bouncer-esque job?

**Comment by warior4356** [comment]
@AffectedArc07 

Regardless of disagreeing with you, I apologize for the tone I took last night. I was frustrated and took it out in a way I shouldn't have. I'm sorry and I will try to keep my discussions more civil in the future.

**Comment by KozackWolf** [comment]
Nanomed machines arent stolen the meds that are layed out to be used are.

You could also say if the orderly is an antag its easier for them in medical.

Self medicating and metafriend healing is a HUGE problem. Essentially you are pushing your way in which tends to interrupt people mid progress bar. Defibs needing time, injections force drinking. Getting interrupted is a huge problem.

The reason medical needs it, is the massive traffic it gets, and the level of import to players, as this keeps tgem in the game. Engineering may have lots of thefts, as well as science, but the traffic they recieve INSIDE their department is a lot lower. Medical they are in the heart of the department and without some semblence of order, it makes treatments take longer, this could make you run out of the 5 minute defib window fairly easily.

The biggest issue I have though, is that there is no way to keep control in medbay. The ether or neurotoxin syringe gun combo is too slow and unreliable, and good luck hitting the right target when chaos is occuring. (Also need an SoP update to allow ethering trespassers if its not in there, as the amount of bitching you get for ethering people is high)

All in all, Maybe orderlies arent right, but there is a real problem with medical traffic, metafriend healing/surgery(i think ooc rules need placed for this to be more strict, trespassing, and theft. I think as long as there is a dialog for change, we can reach a positive outcomec. Please come up with something to help the situation. Hell, I like the idea of a security checkpoint, or, turrents set to tase for anyone who doesnt have a service ticket.(i like the idea like having a ticket machinr for patients served) Just... something needs to change to give medical more control of their department. Please.

**Comment by AffectedArc07** [comment]
> Self medicating and metafriend healing is a HUGE problem. Essentially you are pushing your way in which tends to interrupt people mid progress bar. Defibs needing time, injections force drinking. Getting interrupted is a huge problem.

If metafriending is the issue, thats something that should be solved administratively, instead of creating a medical bouncer.

**Comment by Kiyahitayika** [comment]
" The ether or neurotoxin syringe gun combo is too slow and unreliable, and good luck hitting the right target when chaos is occuring. (Also need an SoP update to allow ethering trespassers if its not in there, as the amount of bitching you get for ethering people is high)"

The reason it's not in SOP is because it's totally _against_ SOP. The procedure you're to follow is to call Security. 

The BIGGEST issue of theft is the stealing of the OR tools by antags/greytiders who are Medstaffers at the beginning of rounds. They spawn in Medbay or even the ORs and just yeet those tools outta there. Sure, you could ask Cargo, but they don't have the points. RnD? They don't have the tech levels. Maybe you're lucky and one set of OR tools were left behind. 

If theft of items like these are the issue, I don't think an "Orderly" is the answer. And, "orderly" isn't even the right _term_ for the role. Hospitals and clinics use "Public Safety" officers. PR a new Security role for this instead. Someone to be on call or on-site for Medbay. When crap hits the fan, they can be called out to deal with crises. They ultimately report to the HoS, but they listen to the instructions of the doctors and CMO. Even if something against Space Law is occurring, it's up to the Doctors to ask them (within reason - obviously if MURDER IS HAPPENING, STEP IN, YOU SKREK!). Just an idea.

**Comment by Leanfrog** [comment]
> The BIGGEST issue of theft is the stealing of the OR tools by antags/greytiders who are Medstaffers at the beginning of rounds. They spawn in Medbay or even the ORs and just yeet those tools outta there. Sure, you could ask Cargo, but they don't have the points. RnD? They don't have the tech levels. Maybe you're lucky and one set of OR tools were left behind.

This bit about Cargo and RnD is wrong, you can just make surgery tools in an autolathe, no research or cargo points required, Cargo already starts with an autolathe and sheets of material ready for use

**Comment by procdrone** [comment]
@Kiyahitayika 

Surgery tools can be printed in the autolathe making their availability very easy.

The only difference for the surgery crate is that it includes sleep gas and medical mask.

**Comment by Oyu-De** [comment]
Wait, someone that has a sole purpose to keep medbay clean and keep the order? Impossible

**Review by Kyep — COMMENTED** [review]
I was actually somewhat in favor of this PR, until I heard the "_People shouldnt self heal. Full stop._" comment.
That comment, alone, somewhat turned me against this PR. Because now I worry its gonna be used to stop anyone self-treating, no matter how incompetent, absent or just useless the doctors are being.

**Comment by Necaladun** [comment]
I think Kyet's worry there is a bit over the top and will come down to the individual orderly and the culture that exists around it.

There's clearly a call for and desire for someone to help keep the peace in medbay. Calling sec generally isn't too viable with the time it takes them to get there, and how minor a lot of the crimes are. 

All the specifics of it can be tweaked (SoP, Equipment, etc), but I'd like to see how it plays out. We can all give horror stories about medbay and sec and the greytide, but I don't want to make a call without seeing it in action a bit.

**Comment by warior4356** [comment]
@Kyep I apologize for being hyperbolic there, I got a little frustrated and heated when that conversation happened. 

People should be allowed to give themselves meds if there are no doctors or meds are left out for people to use. What I feel people shouldn't be doing is ignoring doctors and just barging into medbay. In my mind you shouldn't be medicating yourself unless it was either given to you by a doctor or there are no doctors to treat you.

**Comment by Kyep** [comment]
People absolutely should be allowed to walk into medbay and use basic supplies themselves. 

If the orderly job is going to stop people doing basic stuff, like throwing themselves into a cryopod to benefit from cryox, then I'll probably put my objection on it. I haven't yet, but that sort of stuff would instantly get an objection from me.

There's a reasonable limit to self-treatment. Taking meds you need from the public fridge is always OK. Taking meds you need from the internal fridge is okay if the door to it is left open. Running into medchem to make yourself salt to cure a deadly virus if the chemists aren't present, or aren't doing it, is okay, because you may die or suffer crippling injuries otherwise. Breaking into medchem when chemists are right there doing their jobs, is not okay. Breaking in to make yourself meth is not okay. Breaking in because you were bored is not okay. All of those last things should get you thrown out. But to say that just because medical staff are around, you can NEVER self treat, is not okay.

Half the issue is the map. Our current map is just quite simply not set up for triage, at all. It is not set up to have a flow of diagnosis->treatment. It forces an unholy amount of traffic to congregate in one small area (west of cryopods), when that traffic SHOULD be split up across multiple patient rooms.

Also, I'm really not a fan of orderly, or any other job, including chef, having martial arts. Even area-limited martial arts.
I'd much rather they had a baton or something that could be disarmed or confiscated if they turn out to be terrible, rather than a permanently un-disarmable weapon.

SOP 5, the front desk bit, and escorting people, is not really practical, ever. What's the difference between arresting someone and holding them until sec arrives? None. I'm also against letting orderlies restrain patients. They should be allowed to eject patients from medical, but NOT to force them to undergo treatment unless they're an abductor victim or shadowling thrall. I'm not sure if they should be allowed to hold anyone until sec arrives.

The job needs a sechud icon, too.

It is good that they report to the CMO, who is empowered to fire them if they're terrible.

All mysql-affecting changes, including anything that touches player preferences, needs to be removed from this PR before I can testmerge it. Yeah, that means it can't show up on the job prefs screen - it has to be latejoin-only.

**Comment by KozackWolf** [comment]
Would it be okay to hold onto a version of the PR with the untestmergable stuff. And after its been tested, put it in as a viable starter? That way if its changed to equipment like a baton. It cant be stolen by doctors, or other non orderlies, and just starts in their backpack. Cause if thats in a med access locker, someone else could easily swipe it. Unless we add specific ID allowances to a specifoc locker for orderlies and cmo only.

**Comment by variableundefined** [comment]
Closing due to two maintainers objecting

**Pull request closed without merging**